src/contracts/core/AllocationManager.sol

```diff
@@ -59,28 +59,26 @@ contract AllocationManager is
     function slashOperator(
         address avs,
         SlashingParams calldata params
-    ) external onlyWhenNotPaused(PAUSED_OPERATOR_SLASHING) {
-        // Check that the msg.sender can call - we don't use a modifier to avoid `stack too deep` errors
-        require(_checkCanCall(avs), InvalidCaller());
-        require(0 < params.wadToSlash && params.wadToSlash <= WAD, InvalidWadToSlash());
+    ) external onlyWhenNotPaused(PAUSED_OPERATOR_SLASHING) checkCanCall(avs) {
+        // // Check that the msg.sender can call - we don't use a modifier to avoid `stack too deep` errors
+        // require(_checkCanCall(avs), InvalidCaller());
 
         // Check that the operator set exists and the operator is registered to it
         OperatorSet memory operatorSet = OperatorSet(avs, params.operatorSetId);
         bool isRegistered = _isRegistered(params.operator, operatorSet);
         require(_operatorSets[operatorSet.avs].contains(operatorSet.id), InvalidOperatorSet());
         require(isRegistered, NotMemberOfSet());
 
-        uint256 length = _operatorSetStrategies[operatorSet.key()].length();
-        IStrategy[] memory strategiesSlashed = new IStrategy[](length);
-        uint256[] memory wadSlashed = new uint256[](length);
+        uint256[] memory wadSlashed = new uint256[](params.strategies.length);
 
         // For each strategy in the operator set, slash any existing allocation
-        for (uint256 i = 0; i < length; i++) {
+        for (uint256 i = 0; i < params.strategies.length; i++) {
+            // Check that `wadToSlash` is within acceptable bounds.
+            require(0 < params.wadsToSlash[i] && params.wadsToSlash[i] <= WAD, InvalidWadToSlash());
+
             // 1. Get the operator's allocation info for the strategy and operator set
-            IStrategy strategy = IStrategy(_operatorSetStrategies[operatorSet.key()].at(i));
```

Comment:
So we need to now require the strategy passed in is in the operatorSet because its no longer read from the enumerable set in storage